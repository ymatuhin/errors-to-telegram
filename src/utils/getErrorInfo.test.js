const getErrorInfo = require('./getErrorInfo.js').default;
const { corsError } = require('../config.js');

test('getErrorInfo', () => {
  const fn = () => {};
  const error = new Error('demo error');
  const errorEvent = new ErrorEvent(error);

  expect(getErrorInfo()).toEqual({ message: undefined });
  expect(getErrorInfo(null)).toEqual({ message: 'null' });
  expect(getErrorInfo(NaN)).toEqual({ message: 'null' });
  expect(getErrorInfo(true)).toEqual({ message: 'true' });
  expect(getErrorInfo(false)).toEqual({ message: 'false' });
  expect(getErrorInfo([])).toEqual({ message: '[]' });
  expect(getErrorInfo([1, 2, 3])).toEqual({ message: '[1,2,3]' });

  expect(getErrorInfo(fn)).toEqual({ message: undefined });
  expect(getErrorInfo(error).message).toMatchSnapshot();
  expect(getErrorInfo(error).message).toBeTruthy();
  expect(getErrorInfo(error).stack).toBeTruthy();

  expect(getErrorInfo(errorEvent).message).toMatchSnapshot();
  expect(getErrorInfo(errorEvent).message).toBeTruthy();

  expect(getErrorInfo({ message: 'Script error' }).message).toBe(corsError);
  expect(getErrorInfo({ message: 'Script error.' }).message).toBe(corsError);

  expect(
    getErrorInfo({ filename: 'demo.html', lineno: 1, colno: 2 }).file,
  ).toBe('demo.html:1:2');
});
