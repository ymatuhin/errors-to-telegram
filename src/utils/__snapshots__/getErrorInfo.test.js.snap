// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getErrorInfo 1`] = `"demo error"`;

exports[`getErrorInfo 2`] = `"Error: demo error"`;
